dlc-wallet/src/main/scala/org/bitcoins/dlc/wallet/DLCWallet.scala

```diff
@@ -559,33 +354,50 @@ abstract class DLCWallet extends Wallet with AnyDLCHDWalletApi {
         timeouts = timeouts
       )
 
+      oracleParamsOpt = OracleInfo.getOracleParamsOpt(contractInfo.oracleInfo)
+
       dlcDb = DLCDb(
-        paramHash = paramHash,
+        dlcId = dlcId,
         tempContractId = offer.tempContractId,
         contractIdOpt = None,
+        protocolVersion = 0,
         state = DLCState.Offered,
         isInitiator = true,
         account = account.hdAccount,
         keyIndex = nextIndex,
-        oracleSigsOpt = None,
+        changeIndex = chainType,
         fundingOutPointOpt = None,
         fundingTxIdOpt = None,
         closingTxIdOpt = None,
-        outcomesOpt = None,
-        oraclesUsedOpt = None
+        aggregateSignatureOpt = None
       )
 
-      dlc <- dlcDAO.create(dlcDb)
+      contractDataDb = DLCContractDataDb(
+        dlcId = dlcId,
+        oracleThreshold = contractInfo.oracleInfo.threshold,
+        oracleParamsTLVOpt = oracleParamsOpt,
+        contractDescriptorTLV = contractInfo.contractDescriptor.toTLV,
+        contractMaturity = timeouts.contractMaturity,
+        contractTimeout = timeouts.contractTimeout,
+        totalCollateral = contractInfo.totalCollateral,
+        feeRate = feeRate,
+        fundOutputSerialId = fundOutputSerialId
+      )
 
-      dlcOfferDb = DLCOfferDbHelper.fromDLCOffer(offer)
+      _ <- dlcDAO.create(dlcDb)
```

Comment:
for the future db transaction issue, this stuff looks like it should be included there